OpenRA.Game/Server/Server.cs

```diff
@@ -304,7 +304,7 @@ void ValidateClient(Connection newConn, string data)
 				}
 
 				if (client.Slot != null)
-					SyncClientToPlayerReference(client, Map.Players[client.Slot]);
+					SyncClientToPlayerReference(client, Map.GetPlayers()[client.Slot]);
```

Comment:
You really should cache this instead of regenerating throw-away player definitions each time somebody joins.